src/search.cpp

```diff
@@ -387,7 +391,33 @@ void MainThread::search() {
       for (Thread* th : Threads)
           if (   th->completedDepth > bestThread->completedDepth
               && th->rootMoves[0].score > bestThread->rootMoves[0].score)
-              bestThread = th;
+              longestPVThread = bestThread = th;
+
+      if (bestThread->rootMoves[0].pv.size() < minPlies) {
+          for (Thread* th : Threads) {
+              // Look for the best thread that meets the minimum move criteria.
+              // Don't take a thread unless it's within the appropriate range of score
+              // eval. 
+              if (th->rootMoves[0].pv.size() <= longestPVThread->rootMoves[0].pv.size()) {
+                  continue;
+              }
+              bool possiblePVReplacement = true;
+              for (unsigned int i = 0; i < longestPVThread->rootMoves[0].pv.size(); ++i) {
+                  if (th->rootMoves[0].pv[i] != longestPVThread->rootMoves[0].pv[i]) {
```

Comment:
Would it not be sufficient to compare the PV of the thread with `bestThread`? By comparing with `longestPVThread` the result depends on the order in which the threads are looped over in case there is more than one thread that meets all criteria, but where the PVs are different (while the first moves are agreeing with the PV of `bestThread`).